src/types/entities/channel.rs

```diff
@@ -155,14 +158,40 @@ pub struct PermissionOverwrite {
 }
 
 
-#[derive(Debug, Serialize_repr, Deserialize_repr, Clone, PartialEq, Eq, PartialOrd)]
+#[derive(Debug, Serialize_repr, Clone, PartialEq, Eq, PartialOrd)]
 #[repr(u8)]
 /// # Reference
 pub enum PermissionOverwriteType {
     Role = 0,
     Member = 1,
 }
 
+impl From<u8> for PermissionOverwriteType {
+    fn from(v: u8) -> Self {
+        match v {
+            0 => PermissionOverwriteType::Role,
+            1 => PermissionOverwriteType::Member,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl FromStr for PermissionOverwriteType {
+    type Err = std::num::ParseIntError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        s.parse::<u8>().map(PermissionOverwriteType::from)
+    }
+}
```

Comment:
Er, that's not how it's serialized actually

That is how one would _expect_ give discord's track record, but its encoded as `"role"` or `"member"`, rather than `"0"` or `"1"`